fix: download csv column order

rephus opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

Closes: https://github.com/lightdash/lightdash/issues/7753


### Description:
<!-- Add a description of the changes proposed in the pull request. -->


![Screenshot from 2023-11-07 15-34-47](https://github.com/lightdash/lightdash/assets/1983672/05b93f6f-542b-473d-bfe8-67ca1119031a)
![Screenshot from 2023-11-07 15-24-21](https://github.com/lightdash/lightdash/assets/1983672/8366d3df-167a-43cc-bad5-c712f1e5ccb3)



<!-- Even better add a screenshot / gif / loom -->

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

rephus commented on packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx:
```diff
@@ -108,52 +109,18 @@ const VisualizationCard: FC<{
-                const columnsConfig =
-                    unsavedChartVersion.chartConfig.config?.columns;
-                filteredColumns =
-                    unsavedChartVersion?.metricQuery.dimensions.filter(
```
the old logic was wrong, it was not considering `visible` metrics

ZeRego commented on packages/frontend/src/utils/csvUtils.ts:
```diff
@@ -0,0 +1,39 @@
+        return {
+            metricQuery: {
+                ...metricQuery,
+                metrics: metricQuery.metrics.filter(filterFieldNotVisible),
+                dimensions: metricQuery.dimensions.filter(
+                    filterFieldNotVisible,
+                ),
```
I don't think we should manipulate the metricQuery values.
What happens if a table calculation depends on the dimension/metric?

rephus replied:
yeah, great point, instead we should pass the table config and hide the columns on the csv generation, right ? 


I can simply remove the filtering to fix https://github.com/lightdash/lightdash/issues/7753 but not https://github.com/lightdash/lightdash/issues/7702

rephus commented:
Reverted "filter" code , removed old filtering to fix sorting

![image](https://github.com/lightdash/lightdash/assets/1983672/6b31f400-f0d7-4a07-97d2-2aa81aa9134c)

Fixing visibility will be done in a separate PR
